From 4ee1c4a041c6b956ae56532b85b8405603ba940a Mon Sep 17 00:00:00 2001
From: Santi Villalba <sdvillal@gmail.com>
Date: Fri, 15 May 2026 16:13:41 +0200
Subject: [PATCH] Use system PATH to find tools (in CONDA_PREFIX)

---
 python/triton/knobs.py | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/python/triton/knobs.py b/python/triton/knobs.py
index b55bd3de32..d49d523e20 100644
--- a/python/triton/knobs.py
+++ b/python/triton/knobs.py
@@ -4,6 +4,7 @@
 import importlib
 import os
 import re
+import shutil
 import subprocess
 import sysconfig
 import pathlib
@@ -205,12 +206,22 @@ def get(self) -> NvidiaTool:
     def transform(self, path: str) -> NvidiaTool:
         # We still add default as fallback in case the pointed binary isn't
         # accessible.
-        if path is not None:
-            paths = [path, self.default_path]
-        else:
-            paths = [self.default_path]
+        paths = [path, self.default_path, shutil.which(self.binary)]
+
+        # Conda-forge triton's package:
+        # - depends on recent enough cuda-nvcc-tools, providing suitable ptxas for blackwell.
+        # - does not ship triton's custom ptxas-blackwell.
+        # This heuristic allows conda-forge's triton to work out-of-the-box in blackwell systems.
+        # Lowest priority means no behavior change in other circumstances.
+        # Fixes https://github.com/conda-forge/triton-feedstock/issues/67
+        if str(self.binary).endswith('ptxas-blackwell'):
+            if os.environ.get("CONDA_PREFIX") is not None:
+                conda_path = os.path.join(os.environ["CONDA_PREFIX"], "bin", "ptxas")
+                paths.append(conda_path)
 
         for path in paths:
+            if path is None:
+                continue
             if tool := NvidiaTool.from_path(path):
                 return tool
 
